app/controllers/posts_controller.rb

```diff
@@ -3,6 +3,11 @@ class PostsController < ApplicationController
   skip_authorization_check only: [:create]
   def index
     @posts_by_author = Post.includes(:author).group_by(&:author)
+    serialized_posts = {}
+    @posts_by_author.each do |author, post|
+      serialized_posts[author] = post.as_json(only: %i[Text Title])
+    end
+    render json: serialized_posts
   end
 
```

Comment:
- [ ] Kindly fix the error that is preventing the user from creating a post via  API.
![image](https://github.com/ismayito/Rails-blog-app/assets/81932478/cff84347-5eed-4474-940d-09e36f4cba1a)